node/forest_libp2p/bitswap/src/request_manager.rs

```diff
@@ -143,13 +142,41 @@ impl BitswapRequestManager {
                 }
             }
 
-            if let Ok(()) = block_saved_rx.recv_timeout(BITSWAP_BLOCK_REQUEST_INTERVAL) {
+            if let Ok(data) = block_saved_rx.recv_timeout(BITSWAP_BLOCK_REQUEST_INTERVAL) {
                 success = true;
+                block_data = data;
             }
         }
 
         if !success {
-            success = block_saved_rx.recv_deadline(deadline).is_ok();
+            if let Ok(data) = block_saved_rx.recv_deadline(deadline) {
+                success = true;
+                block_data = data;
+            }
+        }
+
+        if let Some(data) = block_data {
+            success = match Block::new(cid, data) {
+                Ok(block) => match store.insert(&block) {
+                    Ok(()) => {
+                        metrics::message_counter_inbound_response_block_update_db().inc();
+                        true
+                    }
+                    Err(e) => {
+                        metrics::message_counter_inbound_response_block_update_db_failure().inc();
+                        warn!(
+                            "Failed to update db: {e}, cid: {cid}, data: {:?}",
+                            block.data()
+                        );
+                        false
+                    }
+                },
+                Err(e) => {
+                    // TODO: log data
```

Comment:
Do we have an issue for this?

Reply:
It will require copying the data because of ownership thing, and it might not be a good idea to print a up-to-2MB byte array so let me just remove this TODO